tests/e2e/test_app.py

```diff
@@ -52,9 +53,16 @@ def index_name(testrun_uid):
 
 @pytest.fixture(scope="module", autouse=True)
 def knowledge_base(index_name):
-    pinecone.init()
     kb = KnowledgeBase(index_name=index_name)
-    kb.create_canopy_index(indexed_fields=["test"])
+
+    # System and E2E tests are running in parallel and try to create
+    # indexes at the same time.
+    # DB raises an exception when we create two indexes at the same time.
+    # In order to avoid the exception, we create the index here five seconds
+    # later than the system tests.
+    # TODO: Remove the sleep after the DB is fixed.
+    time.sleep(5)
```

Comment:
I think it's better to wrap this with a `@retry(stop_after_attempt=1, wait_random=5)` then a naive `sleep`.  
The `sleep` will only work for a single `pytest` run on the same machine - but in reality we run CI workflows in a matrix, so many jobs are running in parallel.